kernel/Base/MeshManipulatorBase.cpp

```diff
@@ -90,6 +90,12 @@ MeshManipulatorBase::MeshManipulatorBase(const ConfigurationData* config,
 
 MeshManipulatorBase::MeshManipulatorBase(const MeshManipulatorBase& other) =
     default;
+
+Zone* MeshManipulatorBase::addZone(std::string name) {
+    std::size_t index = zones_.size();
+    zones_.emplace_back(name, index);
+    return &zones_[index];
```

Comment:
```suggestion
    zones_.emplace_back(name, zones_.size());
    return &zones_.back();
```
maybe better, but less clear whar is put into the `emplace_back`

Reply:
The `&zones_.back()` is quite nice, so I'll use that